src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -899,6 +909,18 @@ public void Convert(in BL src, ref SB dst)
         public void Convert(in DT src, ref SB dst) { ClearDst(ref dst); dst.AppendFormat("{0:o}", src); }
         public void Convert(in DZ src, ref SB dst) { ClearDst(ref dst); dst.AppendFormat("{0:o}", src); }
         #endregion ToStringBuilder
+        #region ToBL
+        public void Convert(in R8 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in R4 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in I1 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in I2 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in I4 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in I8 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in U1 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in U2 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in U4 src, ref BL dst) => dst = src > 0 ? true : false;
+        public void Convert(in U8 src, ref BL dst) => dst = src > 0 ? true : false;
```

Comment:
From my understanding of #2628, it seems we don't want auto-conversion or any auto magics. Here is a message from Tom,
```
We do not do "auto-conversion" for anything in ML.NET because (1) conversion is the job of transforms and (2) adding the smarts "anywhere" requires that we add them "everywhere," which is a maintainability nightmare. (In fact, it has proven to be practically impossible.)
```